src/borg/helpers.py

```diff
@@ -1404,12 +1417,31 @@ def __init__(self):
             if self.logger.level == logging.NOTSET:
                 self.logger.setLevel(logging.WARN)
             self.logger.propagate = False
+        self.emit = self.logger.getEffectiveLevel() == logging.INFO
```

Comment:
the propagation for the progress logger works so:

- if --progress is _not_ set, then the setup_implied_logging code does nothing. So the immediate logger level is NOTSET, this is checked in the line above, and sets the immediate level to WARN
- if --progress _is_ set, then setup_implied_logging code sets INFO immediate level on the progress logger
- note that the progress logger does _not_ propagate, ie. getEffectiveLevel() == .level -- I could've written it directly that way
- progress logging happens at INFO level
- so if the logger level is INFO, then --progress is turned on, otherwise it is not.